spec/aruba/api/core_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'spec_helper'
+require 'aruba/api'
+require 'fileutils'
+
+describe Aruba::Api::Core do
+  include_context 'uses aruba API'
+
+  describe '#expand_path' do
+    context 'when file_name is given' do
+      it { expect(@aruba.expand_path(@file_name)).to eq File.expand_path(@file_path) }
+    end
+
+    context 'when file_path is given' do
+      it { expect(@aruba.expand_path(@file_path)).to eq @file_path }
+    end
+
+    context 'when path contains "."' do
+      it { expect(@aruba.expand_path('.')).to eq File.expand_path(aruba.current_directory) }
+    end
+
+    context 'when path contains ".."' do
+      it { expect(@aruba.expand_path('path/..')).to eq File.expand_path(File.join(aruba.current_directory)) }
+    end
+
+    context 'when path is nil' do
+      it { expect { @aruba.expand_path(nil) }.to raise_error ArgumentError }
+    end
+
+    context 'when path is empty' do
+      it { expect { @aruba.expand_path('') }.to raise_error ArgumentError }
+    end
+
+    context 'when dir_path is given similar to File.expand_path ' do
```

Comment:
Is this not covered by the other tests; it's a bit tricky for a newbie to see though, so ignore me if I'm wrong.

Also rogue space at the end